docs/python_docs/python/tutorials/getting-started/crash-course/2-create-nn.md

```diff
@@ -0,0 +1,558 @@
+collection of off-the-shelf models e.g. RESNET, BERT etc. Some of the common
+architectures include:
+
+- [Gluon CV model zoo](https://gluon-cv.mxnet.io/model_zoo/index.html)
+
+- [Gluon NLP model zoo](https://gluon-nlp.mxnet.io/model_zoo/index.html)
+
+Lets look at an example
+
+```{.python .input  n=117}
+from mxnet.gluon import model_zoo
+
+net = model_zoo.vision.resnet50_v2(pretrained=True)
+net.hybridize()
+
+dummy_input = np.ones(shape=(1,3,224,224))
+output = net(dummy_input)
+
+output.shape
+        
+```
+
+```{.json .output n=117}
+[
+ {
+  "data": {
+   "text/plain": "(1, 1000)"
+  },
+  "execution_count": 117,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
+
+## Deciding the paradim for your NN
+
+MXNet includes two types of programming paradigms:
+
+1. Imperative API - Gluon that is very user friendly and allows for quick
+prototyping, easy debugging and natural control flow for people familiar with
+python programming.
+2. Symbolic or Declarative API for creating static graphs with low level
+optimizations on operators and allows for support in multiple languages, cross-
+compatibility with other frameworks (via ONNX) and as a backend for other high
+level APIs like keras. However, it’s less flexible because any logic must be
+encoded into the graph as special operators like scan, while_loop and cond. It’s
+also hard to debug.
+
+So how can we make use of Symbolic API while getting the flexibility of an
+Imperative API to quickly protype and debug??
+
+Enter **HybridBlocks**
+
+Each HybridBlock can run fully imperatively defining their computation with real
+functions acting on real inputs. But they’re also capable of running
+symbolically, acting on placeholders. Gluon hides most of this under the hood so
+you’ll only need to know how it works when you want to write your own layers.
+
+```{.python .input  n=122}
+net_hybrid_seq = nn.HybridSequential()
+
+net_hybrid_seq.add(nn.Dense(5,in_units=3,activation='relu'),
+        nn.Dense(25, activation='relu'),
+        nn.Dense(2)
+       )
+
+net_hybrid_seq
+```
+
+```{.json .output n=122}
+[
+ {
+  "data": {
+   "text/plain": "HybridSequential(\n  (0): Dense(3 -> 5, Activation(relu))\n  (1): Dense(-1 -> 25, Activation(relu))\n  (2): Dense(-1 -> 2, linear)\n)"
+  },
+  "execution_count": 122,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
+
+To compile and optimize the `HybridSequential`, we can then call its `hybridize
+method`.
+
+```{.python .input  n=124}
+net_hybrid_seq.hybridize()
+```
+
+Performance
+
+To get a sense of the speedup from hybridizing, we can compare the performance
+before and after hybridizing by measuring in either case the time it takes to
+make 1000 forward passes through the network.
+
+```{.python .input  n=137}
+from time import time
+
+def bench(net, x):
+    start = time()
+    for i in range(1000):
+        y = net(x)
+    return time() - start
```

Comment:
You need to do one forward pass before starting to record time, to ensure that deferred parameter initialization has already happened.